gameobjects/gameobject.cpp

```diff
@@ -44,19 +26,29 @@ void GameObject::SetInstanceName(const std::string& name)
     mInstanceName = name;
 }
 
-void GameObject::SetTimeout(std::chrono::duration<double> timeout, std::function<void()> callback)
+std::vector<Behavior*> GameObject::GetBehaviorComponents() const
 {
-    Pong::GetInstance().GetTimer().AddTimer(GetID(), timeout, callback);
-}
+    std::vector<Behavior*> behaviors {};
 
-void GameObject::PlaySound(const Sound& sound)
-{
-    Pong::GetInstance().GetAudioMixer().PlaySound(sound);
+    const auto& allBehaviors = Behavior::GetComponents();
+
+    for (auto& behavior : allBehaviors)
+    {
+        if (behavior->GetGameObjectID() == mId)
+        {
+            behaviors.push_back(behavior.get());
+        }
+    }
+
+    return behaviors;
```

Comment:
This is a vector of non-owning behaviors. Be careful about their lifetimes and the person requesting them not using them after they die! 

There isn't a bug here, but it is bug-possible

Reply:
Yeah I have thought about this previously. I think it comes down to the idea that nothing should ever be deleted under the user without them explicitly asking for it. So if two objects collide and one is destroyed, it is on the user to know that it should stop using that object after that point. I have done some things in Unity where certain objects being destroy would callback relevant objects before destruction so they can remove the object from various lists and variables. 

It is nicer in Unity using C# though because any variables pointing to a now deleted object get automatically `null`'d without needed to set it to `null`. So instead of doing callbacks, you can also just put `if (object != null)` around everything and it would work the same